src/main/java/seedu/address/logic/parser/SocialInfoMapping.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.logic.parser;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.social.SocialInfo;
+
+/**
+ * Handles mappings of social related identifiers when parsing SocialInfo
+ */
+public class SocialInfoMapping {
+    private static final String INVALID_SYNTAX_EXCEPTION_MESSAGE = "Invalid syntax for social info";
+    private static final String UNRECOGNIZED_SOCIAL_TYPE_MESSAGE = "Unrecognized social type";
+
+    private static final int SOCIAL_TYPE_INDEX = 0;
+    private static final int SOCIAL_USERNAME_INDEX = 1;
+
+    private static final String FACEBOOK_IDENTIFIER = "facebook";
+    private static final String FACEBOOK_IDENTIFIER_ALIAS = "fb";
+    private static final String INSTAGRAM_IDENTIFIER = "instagram";
+    private static final String INSTAGRAM_IDENTIFIER_ALIAS = "ig";
+
+    /**
+     * Returns the SocialInfo object represented by the input String.
+     * If the input does not represent a valid SocialInfo recognized by
```

Comment:
Just to be consistent:
`@throws IllegalValueException if the input does not represent a valid SocialInfo recognized by the defined mappings`